enclave/sgx/verifier.c

```diff
@@ -347,3 +348,177 @@ oe_result_t oe_verify_qve_report_and_identity(
+    if (!p_qve_report_info)
+    {
+        p_qve_report_info_internal = (sgx_ql_qe_report_info_t*)oe_malloc(
+            sizeof(sgx_ql_qe_report_info_t));
+        if (p_qve_report_info_internal == NULL)
+        {
+            result = OE_OUT_OF_MEMORY;
+            goto done;
+        }
+
+        oe_memset_s(
+            p_qve_report_info_internal,
+            sizeof(sgx_ql_qe_report_info_t),
+            0,
+            sizeof(sgx_ql_qe_report_info_t));
+
+        qve_report_info_size = sizeof(sgx_ql_qe_report_info_t);
+
+        // Generate nonce
+        OE_CHECK(oe_random(&nonce, 16));
+        OE_CHECK(oe_memcpy_s(
+            &p_qve_report_info_internal->nonce,
+            sizeof(sgx_nonce_t),
+            &nonce,
+            sizeof(sgx_nonce_t)));
+
+        // Try to get self target info
+        OE_CHECK(
+            oe_get_report(0, NULL, 0, NULL, 0, &p_self_report, &report_size));
+
+        OE_CHECK(oe_get_target_info(
+            p_self_report,
+            report_size,
+            (void**)(&p_self_target_info),
+            &target_info_size));
+
+        OE_CHECK(oe_memcpy_s(
+            &p_qve_report_info_internal->app_enclave_target_info,
+            sizeof(sgx_target_info_t),
+            p_self_target_info,
+            target_info_size));
+    }
+
+    OE_CHECK(oe_verify_quote_ocall(
+        &retval,
+        format_id,
+        NULL,
+        0,
+        p_quote,
+        quote_size,
+        expiration_check_date,
+        p_collateral_expiration_status,
+        p_quote_verification_result,
+        p_qve_report_info_internal,
+        qve_report_info_size,
+        p_supplemental_data,
+        supplemental_data_size,
+        p_supplemental_data_size_out,
+        collateral_version,
+        p_tcb_info,
+        tcb_info_size,
+        p_tcb_info_issuer_chain,
+        tcb_info_issuer_chain_size,
+        p_pck_crl,
+        pck_crl_size,
+        p_root_ca_crl,
+        root_ca_crl_size,
+        p_pck_crl_issuer_chain,
+        pck_crl_issuer_chain_size,
+        p_qe_identity,
+        qe_identity_size,
+        p_qe_identity_issuer_chain,
+        qe_identity_issuer_chain_size));
+
+    result = (oe_result_t)retval;
+
+    if (result != OE_PLATFORM_ERROR)
+    {
+        if (result != OE_OK)
+        {
+            OE_RAISE_MSG(
+                result,
+                "SGX QvE-based quote verification failed with error 0x%x",
+                result);
+        }
+
+        // Defense in depth
+        if (memcmp(
+                &nonce,
+                &p_qve_report_info_internal->nonce,
+                sizeof(sgx_nonce_t)) != 0)
+        {
+            OE_RAISE_MSG(
+                OE_UNEXPECTED,
+                "Nonce during SGX quote verification has been tampered.\n",
+                NULL);
+        }
+
+        // Call internal API to verify QvE identity
+        OE_CHECK_MSG(
+            oe_verify_qve_report_and_identity(
+                p_quote,
+                (uint32_t)quote_size,
```

Comment:
quote_size is already a uint32_t in line 357 so no cast needed
```suggestion
                quote_size,
```

Reply:
Fixed.